File: src/main/java/vorquel/mod/vderps/item/ItemVD.java
package vorquel.mod.vderps.item;

import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import vorquel.mod.vderps.helper.Ref;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

public class ItemVD extends Item {
	
	public final String itemName;
	
	public ItemVD(String name) {
		super();
		itemName = name;
		setCreativeTab(Ref.VD_TAB);
	}
	
	@Override
	public String getUnlocalizedName() {
		return String.format("item.%s", getShortName());
	}
	
	@Override
	public String getUnlocalizedName(ItemStack itemStack) {
		return getUnlocalizedName();
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public void registerIcons(IIconRegister iconRegister) {
		itemIcon = iconRegister.registerIcon(getShortName());
	}
	
	public String getShortName() {
		return String.format("%s:%s", Ref.MOD_ID.toLowerCase(), itemName);
	}
}

File: src/main/scala/vorquel/mod/vderps/item/ItemVD.scala
package vorquel.mod.vderps.item

import cpw.mods.fml.relauncher.{SideOnly, Side}
import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.item.{Item, ItemStack}
import vorquel.mod.vderps.helper.Ref

class ItemVD(name: String) extends Item {
  val itemName: String = name
  setCreativeTab(Ref.VD_TAB)

  override def getUnlocalizedName: String = String.format("item.%s", getShortName)

  override def getUnlocalizedName(itemStack: ItemStack): String = getUnlocalizedName

  @SideOnly(Side.CLIENT)
  override def registerIcons(iconRegister: IIconRegister) {
    itemIcon = iconRegister.registerIcon(getShortName)
  }

  def getShortName: String = String.format("%s:%s", Ref.MOD_ID.toLowerCase, itemName)
}
